gnovm/stdlibs/std/params.go

```diff
@@ -8,65 +8,128 @@ import (
 	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
 )
 
-// ParamsInterface is the interface through which Gno is capable of accessing
+// ParamsSetterInterface is the interface through which Gno is capable of accessing
 // the blockchain's params.
 //
 // The name is what it is to avoid a collision with Gno's Params, when
 // transpiling.
+
 type ParamsInterface interface {
-	SetString(key, val string)
-	SetBool(key string, val bool)
-	SetInt64(key string, val int64)
-	SetUint64(key string, val uint64)
-	SetBytes(key string, val []byte)
+	SetString(key ParamKey, val string)
+	SetBool(key ParamKey, val bool)
+	SetInt64(key ParamKey, val int64)
+	SetUint64(key ParamKey, val uint64)
+	SetBytes(key ParamKey, val []byte)
+}
+
+type ParamKey struct {
+	Realm  string
+	Prefix string
+	Key    string
+	Type   string
+}
+
+func NewParamKey(m *gno.Machine, prefix, key string, kind string) (ParamKey, error) {
+	// validate key.
+	if err := validate(prefix, key, kind); err != nil {
+		es := err.Error()
+		m.Panic(typedString(es))
+		return ParamKey{}, err
+	}
+	_, realm := currentRealm(m)
+
+	return ParamKey{
+		Realm:  realm,
+		Prefix: prefix,
+		Key:    key,
+		Type:   kind,
+	}, nil
+}
+
+// String representation of ParamKey in the format:
+// <realm>.[prefix:]<key>.<type>
+func (pk ParamKey) String() string {
+	pks := ""
+	if pk.Prefix == "" {
+		//	pks = fmt.Sprintf("%s.%s.%s", pk.Realm, pk.Key, pk.Type)
+		pks = fmt.Sprintf("%s.%s", pk.Realm, pk.Key)
+	} else {
+		pks = fmt.Sprintf("%s.%s:%s", pk.Realm, pk.Prefix, pk.Key)
+	}
+	return pks
 }
```

Comment:
i remember that we said that the format was:

- `pk/vm:gno.land/r/foo.Bar.int` for arbitrary
- `pk/bank:_.Bar.bool` for system keepers

Reply:
Yes, there will be additional changes on this and I will update it soon.

Reply:
I updated it along with the documentation.

The universal format is /pv/<module>:(<realm>".")?<key>.
We will move type to value after this PR is merged.

e,g. 
vm:gno.land/r/foo.Bar is used for arbitrary parameters.

bank:Bar is used for bank module parameters. 
vm.Bar is used for vm module parameters

The /pv/ prefix is the store key added in the parameter keeper before <module>:(<realm>".")?<key>, it is not exposed to outside of parameter keeper to prevent store key conflict.
That is the reason it is not shown here.

Reply:
> "Isn't this PR about introducing a new format? Why not use the one we already know?"

Isn't this PR about introducing a new key format? Why are we introducing something intermediary when we already know the target?

Reply:
@moul Which part is the "something intermediary"? Can we be more specific? 

If you are referring to "move type to value after this PR is merged," it is because there will be additional refactoring and validation to be done. There have already been many changes since the last review. However, these are internal modifications and will not affect the token locking feature or its usage, which is a blocker for test6.